### ISSUE TITLE
Filename Specialchar Problem

### ISSUE THREAD

**Issue by cfreak** [spatie/laravel-medialibrary#120]
Hello, 

just found out, that there is a Problem with Files where for example an # is in the Filename.

So maybe, the medialibrary should filter the filenames, or?

So for a better understanding, i do the following:

```php
$incoming->addMedia($request->file('file'))->toCollection('incoming');
```

Everything works fine, but when i try to show the pdf file in the browser, it can't open it, because there is an # in the filename.

Best Regards

**Comment by freekmurze** [comment]
We probably should pay some extra attention to special characters. I'll improve this the next time I'm working on the package. I'd be happy to accept a PR that does some extra validation.

**Comment by freekmurze** [comment]
A first stab at fixing this is included in v3.11.2 of the package. Reopen this if this problems reoccurs.
